File: AOC2022/day 1/solution.py
file_name = 'input.txt'

file = open(file_name, 'r')
Lines = file.readlines()
file.close


cal = []
sum = 0
for i in range(0,len(Lines)-1):
	if not Lines[i].isspace():
		sum = sum + int(Lines[i])
	else:
		cal.append(sum)
		sum = 0

cal.sort(reverse=True)
print(cal[0])
sum = 0
for i in range(3):
	sum = sum + cal[i]

print(sum)

File: AOC2022/day 1/solution.m
clear all
filename = 'input.txt';
S = readlines(filename);
I =  find(strcmp(S,""));
S(I) = 'K';
s = strjoin(S','sp');
cal_sets = split(s,'K');

cal = cellfun(@get_calorie,cal_sets );

cal =  sort(cal,'descend');
cal(1)
sum(cal(1:3))
function [S] = get_calorie(set_cal)
C = str2double(split(set_cal,'sp'));
S = sum(C(~isnan(C)));
end